### PR TITLE
Add `/consensus_params` RPC endpoint

### PR DESCRIPTION
**Pull request by thanethomson**
Partial fulfillment of #832, and helps towards https://github.com/informalsystems/ibc-rs/issues/1336

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Added entry in `.changelog/`

### CODE REVIEW

**Comment by thanethomson** [comment]
@andynog a quick way of testing this would be to run the `tendermint-rpc` binary from this branch against a validator:

```bash
# Get the latest consensus parameters
cargo run --all-features --bin tendermint-rpc -- -u https://validator-address.com latest-consensus-params

# Get the consensus parameters at height 1234
cargo run --all-features --bin tendermint-rpc -- -u https://validator-address.com consensus-params 1234
```

**Comment by adizere** [comment]
Also for testing, to complement Thane's instructions, we can patch in ibc-rs the main [Cargo.toml](https://github.com/informalsystems/ibc-rs/blob/master/Cargo.toml) to use this branch, and call into the `latest_consensus_params` from Andy's branch.

> tendermint              = { git = "https://github.com/informalsystems/tendermint-rs", branch = "thane/832-consensus-params" }
 tendermint-rpc          = { git = "https://github.com/informalsystems/tendermint-rs", branch = "thane/832-consensus-params" }
 tendermint-proto        = { git = "https://github.com/informalsystems/tendermint-rs", branch = "thane/832-consensus-params" }
 tendermint-light-client = { git = "https://github.com/informalsystems/tendermint-rs", branch = "thane/832-consensus-params" }
 tendermint-testgen      = { git = "https://github.com/informalsystems/tendermint-rs", branch = "thane/832-consensus-params" }

@andynog is there a branch WIP for https://github.com/informalsystems/ibc-rs/issues/1336 already?

**Comment by andynog** [comment]
> @andynog a quick way of testing this would be to run the `tendermint-rpc` binary from this branch against a validator:
> 
> ```shell
> # Get the latest consensus parameters
> cargo run --all-features --bin tendermint-rpc -- -u https://validator-address.com latest-consensus-params

Tested this line:
```
{
  "block_height": "7673671",
  "consensus_params": {
    "block": {
      "max_bytes": "200000",
      "max_gas": "40000000",
      "time_iota_ms": "1000"
    },
    "evidence": {
      "max_age_num_blocks": "1000000",
      "max_age_duration": "172800000000000",
      "max_bytes": "50000"
    },
    "validator": {
      "pub_key_types": [
        "ed25519"
      ]
    }
  }
}
```

And this is the response from the server using the RPC endpoint

```
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "block_height": "7673671",
    "consensus_params": {
      "block": {
        "max_bytes": "200000",
        "max_gas": "40000000",
        "time_iota_ms": "1000"
      },
      "evidence": {
        "max_age_num_blocks": "1000000",
        "max_age_duration": "172800000000000",
        "max_bytes": "50000"
      },
      "validator": {
        "pub_key_types": [
          "ed25519"
        ]
      },
      "version": {}
    }
  }
}
```

Most seems OK, the only difference I see is that the RPC outputs a `version` field. I don't need to use it as far as I know for now but just wanted to point that out.

> # Get the consensus parameters at height 1234
> cargo run --all-features --bin tendermint-rpc -- -u https://validator-address.com consensus-params 1234
> ```

This line also seems to work fine:

```
{
  "block_height": "7673671",
  "consensus_params": {
    "block": {
      "max_bytes": "200000",
      "max_gas": "40000000",
      "time_iota_ms": "1000"
    },
    "evidence": {
      "max_age_num_blocks": "1000000",
      "max_age_duration": "172800000000000",
      "max_bytes": "50000"
    },
    "validator": {
      "pub_key_types": [
        "ed25519"
      ]
    }
  }
}
```

File: rpc/src/endpoint/consensus_params.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    fn method(&self) -> crate::Method {
+        crate::Method::ConsensusParams
+    }
+}
+
+impl crate::SimpleRequest for Request {}
+
+/// Consensus parameters response.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct Response {
```
Comment (andynog) [inline rpc/src/endpoint/consensus_params.rs:35]: Should this have a `version` field ? Please see my comments on the issue

**Review by andynog — CHANGES_REQUESTED** [review]
I think overall looks good to me. Had a minor comment about a `version` field

File: rpc/src/endpoint/consensus_params.rs (hunk 1 above)
Comment (romac) [reply rpc/src/endpoint/consensus_params.rs:35]: I don't think it's useful given that this field seems to always be `{}`

**Comment by romac** [comment]
Sorry, about https://github.com/informalsystems/tendermint-rs/commit/be7c665ca9b111b0b7c8417e952527cac6289aa3 and its revert, I failed to notice the `latest_consensus_params` endpoint.

**Review by romac — APPROVED** [review]
All good from my side, we're now using the new endpoint succesfully in https://github.com/informalsystems/ibc-rs/pull/1361 :)

**Review by andynog — APPROVED** [review]

**Pull request merged**